ast/compile.go

```diff
@@ -960,43 +1004,61 @@ func (qc *queryCompiler) WithContext(qctx *QueryContext) QueryCompiler {
 	return qc
 }
 
-func (qc *queryCompiler) WithStageAfter(after string, stage QueryCompilerStage) QueryCompiler {
-	qc.after[after] = append(qc.after[after], stage)
+func (qc *queryCompiler) WithStageAfter(after string, name string, metricName string, stage QueryCompilerStage) QueryCompiler {
```

Comment:
Nit: I would have replaced the stage function with a struct that contains these fields. This would insulate callers from future extensions to the API.

Reply:
We probably should, I'll update the PR with it.

I somehow convinced myself that this was less disruptive (not 100% sure why as I look at it now), but you're right that longer term the struct argument is better.